src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +47,106 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int max = this.max(talks);
+        for (final Talk talk : talks.active()) {
+            max = this.update(talk, max);
+        }
+    }
+
+    /**
+     * Creates an index node in the specified talk node, if it has archive/log
+     * nodes.
+     * @param talk Talk, to which the index node should be appended.
+     * @param max Maximal index value among all talks
+     * @return New maximal value. If an index node was added, it is equal to
+     *  (max+1), otherwise - max.
+     * @throws IOException Thrown in case of XML parsing errors.
+     */
+    private int update(final Talk talk, final int max) throws IOException {
+        final List<String> requests = talk.read().xpath("//request");
+        final List<XML> logs = talk.read().nodes(ARCHIVE_LOG);
+        int newmax = max;
+        int index = this.max(talk);
+        if (index < 1) {
+            index = max;
+        }
+        if (requests.isEmpty() && !logs.isEmpty()) {
+            talk.modify(
+                new Directives().xpath("//talk").add("request")
+                    .attr(INDEX, Integer.toString(index + 1))
+                    .attr("id", this.createRequestId())
+                    .add("type").set(INDEX)
+                    .up()
+                    .add("args")
+            );
+            newmax += 1;
+        }
+        return newmax;
```

Comment:
again, this method is not doing what it is supposed to do. you should set the index of the request to `max`, if such a request exists. if it exists and you sets the `@index`, you return `max+1`. otherwise, you return `max`. this is the logic. you implement something different. we are NOT supposed to create `/talk/request` element in this class!